kitsune/coolsearch/api.py

```diff
@@ -0,0 +1,249 @@
+import json
+from itertools import chain
+
+from django.conf import settings
+from django.http import HttpResponse
+from django.utils.html import escape
+from django.views.generic import View
+
+import bleach
+import jinja2
+from elasticutils.utils import format_explanation
+from tower import ugettext as _, ugettext_lazy as _lazy
+
+from kitsune.coolsearch.forms import SearchForm
+from kitsune.forums.models import ThreadMappingType
+from kitsune.questions.models import QuestionMappingType
+from kitsune.search import es_utils
+from kitsune.search.utils import locale_or_default, clean_excerpt, ComposedList
+from kitsune.sumo.utils import smart_int
+from kitsune.wiki.models import DocumentMappingType
+
+
+EXCERPT_JOINER = _lazy(u'...', 'between search excerpts')
+
+
+class SearchView(View):
```

Comment:
This should probably subclass a pDjango Rest Framework APIView class](https://github.com/tomchristie/django-rest-framework/blob/e11f41ebc4ef088a5849771dfda5a7fba4f82904/docs/api-guide/views.md#class-based-views). That would probably reduce the amount of boiler plate you have to do, and aligns better with how the rest of the APIs on the site are built.